internal/incident/incident.go

```diff
@@ -575,6 +640,115 @@ func (i *Incident) RestoreEscalationStateRules(states []*EscalationState) {
 	}
 }
 
+// RetriggerEscalations tries to re-evaluate the escalations and notify contacts.
+func (i *Incident) RetriggerEscalations() {
+	i.Lock()
+	defer i.Unlock()
+
+	i.runtimeConfig.RLock()
+	defer i.runtimeConfig.RUnlock()
+	defer i.timer.Done()
+
+	if !i.RecoveredAt.IsZero() {
+		// Incident is recovered in the meantime.
+		return
+	}
+
+	ctx := context.TODO()
+	if i.RestoreRecipients(ctx) != nil {
+		// The error is already logged within `i.RestoreRecipients()`
+		return
+	}
+
+	i.evaluateEscalations(i.StartedAt.Add(i.timer.incidentAgeFilter))
+
+	tx, err := i.db.BeginTxx(ctx, nil)
+	if err != nil {
+		i.logger.Errorw("Failed to start time based incident escalations database transaction", zap.Error(err))
+
+		return
+	}
+	defer func() { _ = tx.Rollback() }()
+
+	if i.notifyContacts(ctx, tx, &event.Event{Time: time.Now(), Type: event.TypeInternal}, types.Int{}) != nil {
+		// The error is already logged within `i.notifyContacts()`
+		return
+	}
```

Comment:
Change the event type to `escalation`. Then it also somewhat makes sense to send it to all contacts: for the new ones as a "hey, this incident was escalated to you" and for the old ones "hey, this incident was escalated to more people".